txpool/txpool.go

```diff
@@ -832,27 +829,41 @@ func (p *TxPool) addTx(origin txOrigin, tx *types.Transaction) error {
 		}
 	}
 
-	// check for overflow
-	if slotsRequired(tx) > slotsFree {
-		return ErrTxPoolOverflow
+	slotsAllocated := slotsRequired(tx)
+
+	var slotsFreed uint64
+	if oldTxWithSameNonce != nil {
+		slotsFreed = slotsRequired(oldTxWithSameNonce)
+	}
+
+	var slotsIncreased uint64
+	if slotsAllocated > slotsFreed {
+		slotsIncreased = slotsAllocated - slotsFreed
+		if !p.gauge.increaseWithinLimit(slotsIncreased) {
+			return ErrTxPoolOverflow
+		}
 	}
 
 	// add to index
 	if ok := p.index.add(tx); !ok {
 		metrics.IncrCounter([]string{txPoolMetrics, "already_known_tx"}, 1)
 
+		p.gauge.decrease(slotsIncreased)
```

Comment:
I have already approved, but I think it would be good to have if here:
```
if slotsIncreased > 0 {
   p.gauge.decrease(slotsIncreased)
}
```

Reply:
I fixed it and resolved merge conflicts.